src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/TestEnvironment.cs

```diff
@@ -56,7 +56,10 @@ static bool IsDarwin ()
 		public static bool IsRunningOnHostedAzureAgent {
 			get {
 				string agentNameValue = Environment.GetEnvironmentVariable ("AGENT_NAME");
-				return !string.IsNullOrEmpty(agentNameValue) && agentNameValue.ToUpper ().Contains ("HOSTED");
+				bool hasHostedAgentName = !string.IsNullOrEmpty (agentNameValue) && agentNameValue.ToUpper ().Contains ("AZURE PIPELINES");
+				string serverTypeValue = Environment.GetEnvironmentVariable ("SYSTEM_SERVERTYPE");
+				bool isHostedServerType = !string.IsNullOrEmpty (serverTypeValue) && serverTypeValue.ToUpper ().Contains ("HOSTED");
```

Comment:
Do they have these env vars documented? We could link to?

Reply:
There is _some_ predefined variable documentation here - https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml, but unfortunately it doesn't list the server type variable and it doesn't specify any hosted naming conventions.